### PR TITLE
add test for labels

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by aramase** [comment]
I have already signed the CLA. I see this `You are already authorized to contribute code to this project through your membership` when I sign into https://identity.linuxfoundation.org/projects/cncf

**Comment by aramase** [comment]
@dchen1107 is the dir `/pkg/util/labels` readonly? `pull-kubernetes-verify` is failing because of that.

File: pkg/util/labels/labels_test.go [hunk 1]
```diff
@@ -141,18 +142,28 @@ func TestCloneSelectorAndAddLabel(t *testing.T) {
 			},
 		},
 		{
-			labels:     nil,
-			labelKey:   "foo4",
-			labelValue: "12",
+			labels:           nil,
+			labelKey:         "foo4",
+			labelValue:       "12",
+			matchExpressions: []metav1.LabelSelectorRequirement{{Key: "foo1", Operator: "in"}},
```
Comment (brendandburns) [inline pkg/util/labels/labels_test.go]: Please only add new tests, don't modify existing tests...

**Comment by thockin** [comment]
/approve no-issue

@smarterclayton you marked pkg/util/labels as `.readonly` almost a year ago, with very little comment why.  What is one supposed to do with this?

4f865efdb914b090700147a8aaeee4ce418db40a

**Comment by thockin** [comment]
@fejta you HAVE to make your bot smarter.

**Comment by aramase** [comment]
@thockin there is no reply from @smarterclayton. Any ideas on how I should proceed with this PR?

**Comment by fejta** [comment]
Seems to be working as intended! Just need to figure out how to get people to remove the lgtm sooner -- trying with https://github.com/kubernetes/test-infra/pull/6214

**Comment by brendandburns** [comment]
Ping to @smarterclayton wrt to read-only packages here...

File: pkg/util/labels/labels_test.go [hunk 2]
```diff
@@ -148,11 +149,20 @@ func TestCloneSelectorAndAddLabel(t *testing.T) {
 				"foo4": "12",
 			},
 		},
+		{
+			labels:           nil,
+			labelKey:         "foo4",
+			labelValue:       "89",
+			matchExpressions: []metav1.LabelSelectorRequirement{{Key: "foo1", Operator: "in", Values: []string{"bar1"}}},
```
Comment (thockin) [inline pkg/util/labels/labels_test.go]: why just this one case and not all the interesting cases?

Comment (madhusudancs) [reply pkg/util/labels/labels_test.go]: I am with @thockin here, except that you don't need to be exhaustive. If you can cover some interesting cases to make sure that `matchExpressions` are preserved untouched irrespective of the `labelKey`s and `labelValue`s provided it is sufficient. Particularly, check whether `matchExpressions` are untouched even if you are adding `labelKey`s and `labelValue`s that are already in `matchExpressions`.

Comment (aramase) [reply pkg/util/labels/labels_test.go]: @thockin @madhusudancs sure thing! I will get those added.

Comment (thockin) [reply pkg/util/labels/labels_test.go]: code freeze is coming.  Reassign me if you need me.

**Pull request closed without merging**